gui/default/syncthing/core/pathIsSubDirDirective.js

```diff
@@ -6,7 +6,12 @@ angular.module('syncthing.core')
                 ctrl.$validators.folderPathErrors = function (viewValue) {
                     // This function checks whether ydir is a subdirectory of xdir,
                     // e.g. it would return true if xdir = "/home/a", ydir = "/home/a/b".
+                    // Tildes in both xdir and ydir are expanded for comparison
+                    // so that e.g. xdir = "home/a/", ydir = "~/b" will return true.
                     function isSubDir(xdir, ydir) {
+                        var tildeExpansionRegex = new RegExp("^~\/");
+                        xdir = xdir.replace(tildeExpansionRegex, `${scope.system.tilde}/`);
+                        ydir = ydir.replace(tildeExpansionRegex, `${scope.system.tilde}/`);
```

Comment:
I think you want to use `scope.system.pathSeparator` instead of a hardcoded slash. And perhaps, since we split on that on the line below, perhaps then just check if the first component is `~`?

Reply:
I replaced the hardcoded slash with `scope.system.pathSeperator`. I do think we need to check for `~/` at the beginning of the string since we wouldn't want to expand  `~my-dir/foo`, right?

Reply:
Not asking for any change, just trying to clarify if it's of any interest to you:

How I understand @calmh's suggestion that still works, but no regex required: After splitting, you check that `xdirArr[0] === "~"` and then replace that. For `~my-dir/foo` the array will be `[~my-dir, foo]`, so no match/replace. The small downside: You need to split the expanded tilde again. The upside: No regex or handling of separator types needed, you just get the same behaviour as splitting (rightly or wrongly, but if it's wrong fixing both tilde and splitting consistently would be nice).